tests/fork-change-slot.softhsm

```diff
@@ -0,0 +1,144 @@
+# Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+outdir="output.$$"
+
+echo "Current directory: $(pwd)"
+echo "Source directory: ${srcdir}"
+echo "Output directory: ${outdir}"
+
+mkdir -p $outdir
+
+for i in /usr/lib64/pkcs11 /usr/lib/softhsm /usr/local/lib/softhsm /opt/local/lib/softhsm /usr/lib/x86_64-linux-gnu/softhsm /usr/lib /usr/lib64/softhsm;do
+	if test -f "$i/libsofthsm2.so"; then
+		ADDITIONAL_PARAM="$i/libsofthsm2.so"
+		break
+	else
+		if test -f "$i/libsofthsm.so";then
+			ADDITIONAL_PARAM="$i/libsofthsm.so"
+			break
+		fi
+	fi
+done
+
+if (! test -x /usr/bin/p11tool && ! test -x /usr/local/bin/p11tool);then
+	exit 77
+fi
+
+init_db () {
+	if test -x "/usr/local/bin/softhsm2-util"; then
+		export SOFTHSM2_CONF="$outdir/softhsm-testpkcs11.config"
+		SOFTHSM_TOOL="/usr/local/bin/softhsm2-util"
+	fi
+
+	if test -x "/opt/local/bin/softhsm2-util"; then
+		export SOFTHSM2_CONF="$outdir/softhsm-testpkcs11.config"
+		SOFTHSM_TOOL="/opt/local/bin/softhsm2-util"
+	fi
+
+	if test -x "/usr/bin/softhsm2-util"; then
+		export SOFTHSM2_CONF="$outdir/softhsm-testpkcs11.config"
+		SOFTHSM_TOOL="/usr/bin/softhsm2-util"
+	fi
+
+	if test -x "/usr/bin/softhsm"; then
+		export SOFTHSM_CONF="$outdir/softhsm-testpkcs11.config"
+		SOFTHSM_TOOL="/usr/bin/softhsm"
+	fi
+
+	if test -z "${SOFTHSM_TOOL}"; then
+		echo "Could not find softhsm(2) tool"
+		exit 77
+	fi
+
+	if test -z "${SOFTHSM_CONF}"; then
+		rm -rf $outdir/softhsm-testpkcs11.db
+		mkdir -p $outdir/softhsm-testpkcs11.db
+		echo "objectstore.backend = file" > "${SOFTHSM2_CONF}"
+		echo "directories.tokendir = $outdir/softhsm-testpkcs11.db" >> "${SOFTHSM2_CONF}"
+	else
+		rm -rf $outdir/softhsm-testpkcs11.db
+		echo "0:$outdir/softhsm-testpkcs11.db" > "${SOFTHSM_CONF}"
+	fi
+}
+
+init_card () {
+	PIN="$1"
+	PUK="$2"
+    LABEL="$3"
+
+	echo -n "* Initializing smart card... "
+	${SOFTHSM_TOOL} --init-token --free --label "${LABEL}" --so-pin "${PUK}" --pin "${PIN}" >/dev/null
+	if test $? = 0; then
+		echo ok
+	else
+		echo failed
+		exit 1
+	fi
+}
+
+PIN=1234
+PUK=1234
+
+# Initialize SoftHSM db
+init_db
+
+# Create 2 different tokens
+init_card $PIN $PUK token1
+init_card $PIN $PUK token2
+
+# Generate a key pair in the second token
+p11tool --generate-rsa --bits=2048 --label=pkey --login \
+    "pkcs11:token=token2;pin-value=$PIN" >/dev/null
+
+if test $? != 0;then
+	exit 1;
+fi
+
+# Generate a certificate using the key from the second token
+openssl req -engine pkcs11 -new -key \
+    "pkcs11:token=token2;object=pkey;type=private;pin-value=$PIN" \
+    -keyform engine -out $outdir/cert.crt -x509 -subj "/CN=localhost"
+
+# Run the test program which will stop and wait for a signal (SIGUSR1)
+./fork-change-slot \
+    "pkcs11:token=token2;object=pkey;type=private;pin-value=$PIN" \
+    "$outdir//cert.crt" 2 &
+pid=$!
+
+# Wait the test program to reach the sigwait
+sleep 1
+
+# Remove the first token to change the list of slots
```

Comment:
`to change the order of listed slots` may be more  clear

Reply:
I do not agree on this, since the goal is not to change the order of listed slots, but to change the slot to which the used device (``token2`` in my test) is associated. I would rephrase it to ``"Remove the fist token to change de slot ID associated with token2"``. Do you agree?

Reply:
fine with me